### ISSUE TITLE
Access violation in TGetOptsParams on FreeBSD 12.1

### ISSUE THREAD

**Issue by zamronypj** [fanoframework/fano#12]
on FreeBSD 12.1 with Free Pascal 3.2.0 and binutils 2.33
Compilation with `build.sh` script is is success. 
```
$ BUILD_TYPE=dev ./build.sh
```
But when executable runs, it raises access violation exception on TGetOptParams line 276
```
$ ./bin/app.cgi
An unhandled exception occurred at $000000000042354B:
EAccessViolation: Access violation
  $000000000042354B
  $0000000000459E7B  BUILD,  line 276 of vendor/fano/src/Libs/Cli/Implementations/GetOptsParamsImpl.pas

Heap dump by heaptrc unit of ./bin/app.cgi
52663 memory blocks allocated : 5975448/5975856
52660 memory blocks freed     : 5975256/5975664
3 unfreed memory blocks : 192
True heap size : 1277952 (32 used in System startup)
True free heap : 1277056
Should be : 1277152
Call trace for block $0000000800DF57E0 size 128
  $00000000004146B1
  $000000000042354B
Call trace for block $0000000800D99C00 size 40
  $00000000004146B1
  $000000000042354B
Call trace for block $0000000800D99B00 size 24
  $000000000042354B
```

**Comment by zamronypj** [comment]
Note only FreeBSD, it happens in Ubuntu 18.04 too after upfrage to FPC 3.2.0

**Comment by zamronypj** [comment]
Fix in [2d531e44d3](https://github.com/fanoframework/fano/commit/2d531e44d3d84583ee335e569bf1bc2121cc9257)
See related bug https://bugs.freepascal.org/view.php?id=37233
